src/service/client-report/client-report.service.ts

```diff
@@ -0,0 +1,220 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../../db/prisma.service';
+import { DataCapStatsService } from '../datacapstats/datacapstats.service';
+import { DateTime } from 'luxon';
+import { ProteusShieldService } from '../proteus-shield/proteus-shield.service';
+import { LocationService } from '../location/location.service';
+import { IPResponse } from '../location/types.location';
+
+@Injectable()
+export class ClientReportService {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly dataCapStatsService: DataCapStatsService,
+    private readonly proteusShieldService: ProteusShieldService,
+    private readonly locationService: LocationService,
+  ) {}
+
+  async generateReport(client: string) {
+    const verifiedClientResponse =
+      await this.dataCapStatsService.fetchClientDetails(client);
+
+    const verifiedClientData =
+      this.dataCapStatsService.findPrimaryClientDetails(
+        verifiedClientResponse.data,
+      );
+
+    const storageProviderDistribution =
+      await this.getStorageProviderDistributionWithLocation(client);
+
+    const replicaDistribution = await this.getReplicationDistribution(client);
+
+    const cidSharing = await this.getCidSharing(client);
+
+    await this.prismaService.client_report.create({
+      data: {
+        client: client,
+        client_address: verifiedClientData?.address,
+        organization_name:
+          (verifiedClientData?.name ?? '') +
+          (verifiedClientData?.orgName ?? ''),
+        storage_provider_distribution: {
+          create: storageProviderDistribution?.map(
+            (storageProviderDistribution) => {
+              return {
+                provider: storageProviderDistribution.provider,
+                unique_data_size: storageProviderDistribution.unique_data_size,
+                total_deal_size: storageProviderDistribution.total_deal_size,
+                ...(storageProviderDistribution.location && {
+                  location: {
+                    create: storageProviderDistribution.location,
+                  },
+                }),
+              };
+            },
+          ),
+        },
+        replica_distribution: {
+          create: replicaDistribution,
+        },
+        cid_sharing: {
+          create: cidSharing,
+        },
+      },
+    });
+  }
+
+  private async getStorageProviderDistributionWithLocation(client: string) {
+    const lastWeek = DateTime.now()
+      .toUTC()
+      .minus({ week: 1 })
+      .startOf('week')
+      .toJSDate();
+
+    const clientProviderDistribution =
+      await this.prismaService.client_provider_distribution_weekly.findMany({
+        where: {
+          client: client,
+          week: lastWeek,
```

Comment:
We're interested in total distribution for this client, not just last week.
We need to aggregate all weeks together.

Which I think we don't actually have right now? We can't just sum `unique_data_size` as that wouldn't catch duplicates made in different weeks.

I think we need to add new aggregation, that would just process this for all data, without `week` aggregation.